### PR TITLE
ACS RP Schema template.

### PR DESCRIPTION
**Pull request by amansdhanjal**
ACS RP Schema Template with test file and other changes. Please review and merge this.

Thanks,
Amandeep

### CODE REVIEW

File: schemas/2016-03-30/Microsoft.ContainerService.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        }
+      },
+      "type": "object",
+      "required": [
+        "orchestratorType"
+      ]
+    },
+    "masterProfile": {
+      "properties": {
+        "count": {
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: please update this to support template expression

File: schemas/2016-03-30/Microsoft.ContainerService.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        "ssh"
+      ]
+    },
+    "agentPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "count": {
+          "type": "integer"
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: please update this to support template expression.

File: schemas/2016-03-30/Microsoft.ContainerService.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2016-03-30"
+          ]
+        },
+        "properties": {
+          "properties": {
+            "orchestratorProfile": {
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: please update the properties to support template expression.

Comment (amansdhanjal) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: Done

File: schemas/2016-03-30/Microsoft.ContainerService.json (hunk 1 above)
Comment (amansdhanjal) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: Done

File: schemas/2016-03-30/Microsoft.ContainerService.json [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "count",
+        "dnsPrefix"
+      ]
+    },
+    "publicKey": {
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: this seems not defining the element of an array but the properties of an array, the correct one is using "items".

File: schemas/2016-03-30/Microsoft.ContainerService.json [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+            "$ref": "#/definitions/agentPoolProfile"
+          },
+          {
+            "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+          }
+        ]
+      },
+      "type": "array",
+      "required": [
+        "items"
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: Is "items" a property of the agentPoolProfiles?

Comment (amansdhanjal) [inline schemas/2016-03-30/Microsoft.ContainerService.json]: No. AgentPoolProfiles is basically an array with different properties (specified in agentPoolProfile).

**Pull request merged**